src/pybind/mgr/cephadm/services/mgmt_gateway.py

```diff
@@ -1,14 +1,41 @@
 import logging
-from typing import List, Any, Tuple, Dict, cast
+from typing import List, Any, Tuple, Dict, cast, Optional
 
 from orchestrator import DaemonDescription
 from ceph.deployment.service_spec import MgmtGatewaySpec, GrafanaSpec
 from cephadm.services.cephadmservice import CephadmService, CephadmDaemonDeploySpec, get_dashboard_endpoints
-from cephadm.ssl_cert_utils import SSLCerts
+from mgr_util import build_url
 
 logger = logging.getLogger(__name__)
 
 
+def get_mgmt_gw_internal_endpoint(svc: CephadmService) -> Optional[str]:
```

Comment:
the use of the `svc` object feels off to me. Given you call `_inventory_get_fqdn` it seems like we need the `svc` object to be the mgmt gateway one, not just any. Can this function just be part of the mgmt-gateway class, given you'd just have to pass an instance of the class to call it anyway?

Reply:
moved the function `_inventory_get_fqdn` to the cephadm module and changed all the call to use the new function. I think it makes more sense as this method should be global and provided from the mgr module.